go/go.mod

```diff
@@ -1,16 +1,16 @@
 module github.com/smartcontractkit/data-streams-sdk/go
 
-go 1.22.4
+go 1.24.0
 
 require (
-	github.com/ethereum/go-ethereum v1.14.7
+	github.com/ethereum/go-ethereum v1.16.7
```

Comment:
I can be convinced not to do this change, if that would be bad in any way. The reason it's here is because the old version of GoEth has a couple of CVEs.